frontend/fingertips-frontend/upload/components/organisms/BatchListTable/index.tsx

```diff
@@ -1,10 +1,14 @@
 import { Batch } from '@/generated-sources/ft-api-client';
 import { Table, Button, H4 } from 'govuk-react';
-import React from 'react';
-import { deleteBatch } from '../../forms/IndicatorUploadForm/uploadActions';
+import React, { Dispatch, SetStateAction } from 'react';
+import {
+  ApiResponse,
+  deleteBatch,
+} from '../../forms/IndicatorUploadForm/uploadActions';
 
 type BatchListTableProps = {
   batches: Batch[];
+  deletionState: Dispatch<SetStateAction<ApiResponse>> | undefined;
```

Comment:
I'm not sure `deletionState` is a very clear name for this property, as it's a callback to be used to set the deletion response. How about:
```suggestion
  setDeleteResponse: Dispatch<SetStateAction<ApiResponse>> | undefined;
```